homework_3/barbhaiwala/solution.js

```diff
@@ -0,0 +1,83 @@
+function NaiveCalculator(equation)
+{
+    this.splitTokens = function()
+    {
+        return equation.split(' ');
+    }
+    
+    this.applyOperation = function (operand1, operation, operand2) {
+        var resultOfOperation;
+        switch(operation)
+        {
+            case '+':
+                resultOfOperation = operand1 + operand2;
+                break;
+            case '-':
+                resultOfOperation = operand1 - operand2;
+                break;
+            case '/':
+                resultOfOperation = operand1 / operand2;
+                break;
+            case 'x':
+            case 'X':
+                resultOfOperation = operand1 * operand2;
+                break;
+            case '^':
+                resultOfOperation = Math.pow(operand1, operand2);
+                break;
+        }
+        
+        return resultOfOperation;
+    }
+    
+    this.colorize = function (param)
+    {
+        var retValue;
+        if (param > 0)
+        {
+            retValue = (param.toString() .green);
+        }
+        else if (param < 0)
+        {
+            retValue = (param.toString() .red);
+        }
+        else if (param === 0)
+        {
+            retValue = (param.toString() .yellow);
+        }
+        
+        return retValue;
+    }
+    
+    this.evaluate = function ()
+    {
+        var tokens = splitTokens();
+        var operand1;
+        var currentToken;
+        
+        for(var currentTokenCtr = 0; currentTokenCtr < tokens.length; currentTokenCtr++)
+        {
+            currentToken = tokens[currentTokenCtr]
+            var potentialOperand = parseFloat(currentToken);
+            
+            if (!isNaN(potentialOperand))
+            {
+                operand1 = parseInt(currentToken);
+            }
+            else
+            {
+                operand1 = applyOperation(operand1, currentToken, 
+                    parseFloat(tokens[++currentTokenCtr]));
```

Comment:
Hmm... I would suggest a different approach than incrementing your counter inside the `for`. It means you have a weird sort of double increment situation that may be difficult to reason about. It might be easier to push each element on an array and call `applyOperation` at the end of each loop if the length is 3.